cmd/endpointQuerier/main.go

```diff
@@ -0,0 +1,176 @@
+	tlsVersionGaugeVec.WithLabelValues(endpoints.NamespaceifyString(organizationName)).Set(float64(resp.TLS.Version))
+}
+
+func initializeMetrics(listOfEndpoints endpoints.ListOfEndpoints) {
+	httpCodesGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "AllEndpoints",
+			Name:      "http_request_responses",
+			Help:      "HTTP requests partitioned by OrgName",
+		},
+		[]string{"orgName"})
+
+	responseTimeGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "AllEndpoints",
+			Name:      "http_response_time",
+			Help:      "HTTP response time partitioned by OrgName",
+		},
+		[]string{"orgName"})
+
+	tlsVersionGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "AllEndpoints",
+			Name:      "tls_version",
+			Help:      "TLS version reported in the response header",
+		},
+		[]string{"orgName"})
+
+	fhirVersionGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "AllEndpoints",
+			Name:      "fhir_version",
+			Help:      "FHIR version reported in the conformance statement",
+		},
+		[]string{"orgName"})
+
+	totalUptimeChecksCounterVec = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "AllEndpoints",
+			Name:      "total_uptime_checks",
+			Help:      "Total number of uptime checks partitioned by OrgName",
+		},
+		[]string{"orgName"})
+
+	totalFailedUptimeChecksCounterVec = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "AllEndpoints",
+			Name:      "total_failed_uptime_checks",
+			Help:      "Total number of failed uptime checks partitioned by OrgName",
+		},
+		[]string{"orgName"})
+
+	prometheus.MustRegister(httpCodesGaugeVec)
+	prometheus.MustRegister(responseTimeGaugeVec)
+	prometheus.MustRegister(tlsVersionGaugeVec)
+	prometheus.MustRegister(fhirVersionGaugeVec)
+	prometheus.MustRegister(totalUptimeChecksCounterVec)
+	prometheus.MustRegister(totalFailedUptimeChecksCounterVec)
+
+}
+
+func setupServer() {
+	// Setup hosted metrics endpoint
+	http.Handle("/metrics", promhttp.Handler())
+	var err = http.ListenAndServe(":8443", nil)
+	if err != nil {
+		println("HTTP Request Error: %s", err)
+	}
+}
+
+func main() {
+
+	go setupServer()
+
+	var endpointsFile = os.Args[1]
+	// Data in resources/EndpointSources was taken from https://fhirendpoints.github.io/data.json
+	var listOfEndpoints = endpoints.GetListOfEndpoints(endpointsFile)
+	initializeMetrics(listOfEndpoints)
+
+	var queryCount = 0
+	// Infinite query loop
+	for {
+		for _, endpointEntry := range listOfEndpoints.Entries {
+			// TODO: Distribute calls using a worker of some sort so that we are not sending out a million requests at once
+			var url = endpointEntry.FHIRPatientFacingURI
+			var orgName = endpointEntry.OrganizationName
+			// Long polling stats will be queried for every 6 hours
+			var longPollingInterval = (queryCount%72 == 0)
```

Comment:
where is 72 coming from?

Reply:
When the polling interval is every 5 minutes, (5 minutes * 72) * 1 hour/ 60minutes = 6 hours. This would make it so we are only checking/recording data in the conformance statement once every 6 hours. Since parallel data collection was removed this is running less than once every 6 hours since it takes longer than 5 minutes to do all of the data collection in series

Reply:
what happens when you change the polling interval (which you have several times)? Looks like this needs to be generalized.

Reply:
These values will be added to the .config file